typhon/signal.py

```diff
@@ -56,29 +57,44 @@ def signal_widget(signal, read_only=False):
         # Register signal with plugin
         register_signal(signal)
         chan = channel_name(signal.name, protocol='sig')
-    # Check for enum_strs, if so create a QCombobox
-    if read_only:
-        logger.debug("Creating Label for %s", signal.name)
-        widget = TyphonLabel
-        name = signal.name + '_label'
-    else:
-        # Grab a description of the widget to see the correct widget type
-        try:
-            desc = signal.describe()[signal.name]
-        except Exception:
-            logger.error("Unable to connect to %r during widget creation",
-                         signal.name)
-            desc = {}
-        # Create a QCombobox if the widget has enum_strs
-        if 'enum_strs' in desc:
-            logger.debug("Creating Combobox for %s", signal.name)
-            widget = TyphonComboBox
-            name = signal.name + '_combo'
-        # Otherwise a LineEdit will suffice
+    # Grab a description of the widget to see the correct widget type
+    try:
+        desc = signal.describe()[signal.name]
+    except Exception:
+        logger.error("Unable to connect to %r during widget creation",
+                     signal.name)
+        desc = {}
+    # Unshaped data
+    if desc.get('shape', []) == []:
```

Comment:
```suggestion
    shape = desc.get('shape', [])
    try:
        dimensions = len(shape)
    except TypeError:
        dimensions = 0
    if dimensions == 0:
```

Reply:
In case `shape` has no `len`?

Reply:
It really _shouldn't_ be the case (as it would break the data model), but you also wouldn't want it crashing here unnecessarily in case `None` or a scalar gets through.